homeassistant/components/micropel/binary_sensor.py

```diff
@@ -0,0 +1,100 @@
+"""Support for Micropel Coil and Discrete Input sensors."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import PLATFORM_SCHEMA, BinarySensorEntity
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_BINARY_SENSORS,
+    CONF_DEVICE_CLASS,
+    CONF_NAME,
+    CONF_UNIQUE_ID,
+)
+from homeassistant.helpers import config_validation as cv
+
+from . import BINARY_SENSOR_SCHEMA
+from .const import CONF_BIT_INDEX, CONF_HUB, CONF_PLC, DOMAIN
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_BINARY_SENSORS): vol.All(
+            cv.ensure_list, [BINARY_SENSOR_SCHEMA]
+        ),
+    }
+)
```

Comment:
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Please note that this integration connects to a device or service and another Architecture Decision Record applies that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1:

Reply:
This is the complex integration similar to modbus. I don't think that it is possible to configure it through GUI.

Reply:
@frenck Where I can ask for an exception?

Reply:
I think that is possible, nevertheless, ADR-0007 still applies

Reply:
Yes, it makes sense. @frenck Could you please help me? Do you know which integration use configuration YAML by ADR-0007? Thank you in advance.

Reply:
KNX might be a good example

Reply:
Thank you. I will modify my configuration according to KNX. Could you please reopen my PR and move it to draft state to keep our communication? I would like to prevent discussing it again from beginning with somebody else.

Reply:
Please open a new PR when ADR requirements are met and the communication logic is extracted to a PyPi package.

Reply:
OK, I will. 
Communication logic is already extracted to PyPi, only modification of configuration YAML remains.  
Did I miss something?

Reply:
Aah, that was squashed in a commit. Please don't squash commits!

Reply:
Do you prefer multiple commits in one PR? Every project has different rules and this is my first PR to HASS.

Reply:
Please don't squash commits, it makes it really hard for reviewers to follow the changes that are happening during the review process.

This project squashes commits on merge of the PR (we that for you).

Reply:
Thank you for your help